### PR TITLE
REDDEV-620 expose report title

### PR DESCRIPTION
**Pull request by wsams**
Expose report title in each count card.

Add Database class for handling direct SQL calls.

### CODE REVIEW

File: lib/data.php [hunk 1]
```diff
@@ -53,16 +33,18 @@ function reportExists($reportId, $reportsArray) {
 $reportConfigInstance = new ReportConfig($project_id, $module);
 $config = $reportConfigInstance->getReportConfig();
 $dataDictionary = new DataDictionary(\REDCap::getDataDictionary('array'));
+$db = new Database($rc_connection /* $rc_connection is globabl */);
```
Comment (lawhead) [inline lib/data.php]: The comment has an extra letter in 'global'.

File: lib/data.php [hunk 2]
```diff
@@ -53,16 +33,18 @@ function reportExists($reportId, $reportsArray) {
...
 
 if (isset($_GET['action'])) {
     if ($_GET['action'] === 'getReports') {
-        $returnArray = getReports();
+        $returnArray = $db->getReports($project_id /* $project_id is gloabl */);
```
Comment (lawhead) [inline lib/data.php]: Misspelling in comment.

File: lib/data.php [hunk 3]
```diff
@@ -72,7 +54,7 @@ function reportExists($reportId, $reportsArray) {
         }
     }
 } else {
-    $reportsArray = getReports();
+    $reportsArray = $db->getReports($project_id /* $project_id is gloabl */);
```
Comment (lawhead) [inline lib/data.php]: Here too.

File: lib/data.php (hunk 1 above)
Comment (wsams) [reply lib/data.php]: Good catch, the others were c/p.

File: lib/Database.php [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        $returnArray = array();
+        while (mysqli_stmt_fetch($stmt)) {
+            $returnArray[] = array('reportId' => $reportId, 'title' => $title);
+        }
+        mysqli_stmt_close($stmt);
+        return $returnArray;
+    }
+
+    /**
+     * Queries the REDCap database directory to retrieve the title for the give report id.
```
Comment (aeyates) [inline lib/Database.php]: title for the "given" report id

**Pull request merged**